File: go/hooks/hosts/all.go
// Package hosts registers every axon hook codec. Import it for side
// effects; import an individual host package to register only that one.
package hosts

import (
	"hop.top/axon/hooks"
	"hop.top/axon/hooks/hosts/claude"
	"hop.top/axon/hooks/hosts/codex"
	"hop.top/axon/hooks/hosts/gemini"
	"hop.top/axon/hooks/hosts/opencode"
)

func init() {
	hooks.Register(claude.New())
	hooks.Register(gemini.New())
	hooks.Register(codex.New())
	hooks.Register(opencode.New())
}

File: ts/src/hooks/hosts/all.ts
// Importing this module registers the codecs for every hooked host that
// currently has one (claude, codex, gemini, opencode -- contract notes
// §4). Mirrors hop.top/axon/hooks/hosts's blank-import side effect.
import { register } from "../registry";
import { makeCodec as makeClaudeCodec } from "./claude";
import { makeCodec as makeCodexCodec } from "./codex";
import { makeCodec as makeGeminiCodec } from "./gemini";
import { makeCodec as makeOpencodeCodec } from "./opencode";

register(makeClaudeCodec());
register(makeCodexCodec());
register(makeGeminiCodec());
register(makeOpencodeCodec());
